qiskit_nature/transformers/second_quantization/electronic/seniority_zero_transformer.py

```diff
@@ -0,0 +1,242 @@
+                               `ParticleNumber` instance.
+        """
+        if not isinstance(grouped_property, GroupedElectronicProperty):
+            raise QiskitNatureError(
+                "Only `GroupedElectronicProperty` objects can be transformed by this Transformer, "
+                f"not objects of type, {type(grouped_property)}."
+            )
+
+        particle_number = grouped_property.get_property(ParticleNumber)
+        if particle_number is None:
+            raise QiskitNatureError(
+                "The provided `GroupedElectronicProperty` does not contain a `ParticleNumber` "
+                "property, which is required by this transformer!"
+            )
+        particle_number = cast(ParticleNumber, particle_number)
+
+        self._num_modes = particle_number.num_spin_orbitals // 2
+
+        grouped_property_transformed = self._transform_property(grouped_property)
+
+        return grouped_property_transformed  # type: ignore[return-value]
+
+    def _transform_property(self, prop: Property) -> Property:
+        """Transforms a `Property` object.
+
+        This is a recursive reduction, iterating `GroupedProperty` objects when encountering one.
+
+        Args:
+            property: the property object to transform.
+
+        Returns:
+            The transformed property object.
+
+        Raises:
+            TypeError: if an unexpected Property subtype is encountered.
+        """
+        transformed_property: Property
+        # Code for recursion is copied from ActivateSpaceTransformer
+        if isinstance(prop, GroupedProperty):
+            transformed_property = deepcopy(prop)
+
+            for internal_property in iter(prop):
+                try:
+                    transformed_internal_property = self._transform_property(internal_property)
+                    if transformed_internal_property is not None:
+                        transformed_property.add_property(transformed_internal_property)
+                    else:
+                        transformed_property.remove_property(internal_property)
+                except TypeError:
+                    logger.warning(
+                        "The Property %s of type %s could not be transformed!",
+                        internal_property.name,
+                        type(internal_property),
+                    )
+                    continue
+
+            if len(transformed_property._properties) == 0:
+                transformed_property = None
+
+        elif isinstance(prop, ElectronicEnergy):
+            h_1 = prop.get_electronic_integral(ElectronicBasis.MO, 1).get_matrix()
+            h_2 = prop.get_electronic_integral(ElectronicBasis.MO, 2).get_matrix()
+
+            hr_1, hr_2 = self._restricted_electronic_integrals(h_1, h_2)
+
+            transformed_property = ElectronicEnergy.from_raw_integrals(
+                ElectronicBasis.MO, hr_1, hr_2
+            )
```

Comment:
Technically, `hr_2` has the shape of 1-body terms but will be injected into a `TwoBodyElectronicIntegrals` object here. There is currently no better way of doing this unfortunately, because of the internal structure of the `IntegralProperty` and `ElectronicIntegrals` classes.
These will be refactored in the near future to resolve this temporary workaround.